features/323-login-id/api.md

```diff
@@ -0,0 +1,170 @@
+
++ async requestVerification(loginIDKey: string, loginID: string): Promise<void>;
++ async createLoginID(loginIDKey: string, loginID: string, realm?: string): Promise<void>;
++ async deleteLoginID(loginIDKey: string, loginID: string, realm?: string): Promise<void>;
+
+  async whoami(): Promise<User>;
+```
+
+### Example user object
+```json
+{
+    "user_id": "5bf1e4d2-e1c4-4517-93c7-7dae89261da6",
+    "metadata": {
+        "email": "test@example.com"
+    },
+    "last_seen_at": "...",
+    "created_at": "...",
+    "created_by": "...",
+    "updated_at": "...",
+    "updated_by": "...",
+    "verified": true,
+    "verify_info": {
+        "test@example.com": true
+    },
+    "access_token": "..."
+}
+```
+
+## Use Case Examples
+
+Unless specified, following examples assume using default configuration.
+
+### Signup with single login ID
+```typescript
+const user1 = await signupWithEmail("test@example.com", "12345678");
+// user1.metadata is { "email": "test@example.com" }
+const user2 = await signup({ phone: "+85299999999" }, "12345678");
+// user2.metadata is { "phone": "+85299999999" }
+
+// following should fail:
+await signup({ fingerprint: "ZmluZ2VycHJpbnQ=" }, "12345678"); // not white-listed
+```
+
+### Signup with realms
+Assume `teacher` and `student` are whitelisted realms.
+```typescript
+await signupWithEmail("test@example.com", "12345678", undefined, "teacher");
+await createLoginID("email", "test@example.com", "student");
+// then:
+
+// - following should succeed:
+await loginWithEmail("test@example.com", "12345678", "teacher");
+await loginWithEmail("test@example.com", "12345678", "student");
+
+// - following should fail:
+await loginWithEmail("test@example.com", "12345678"); // realm not matched
+await signupWithEmail("test@example.com", "12345678", undefined, "admin"); // cannot create login ID (use createLoginID instead)
+await signupWithEmail("test@example.com", "12345678"); // cannot reuse login ID for new user
+```
+
+### Signup with multiple login ID
+Assume `email` and `username` are white-listed login ID keys,
+with exactly one username allowed.
+
+```typescript
+const user = await signup([
+    {"email": "test@oursky.com"},
+    {"email": "test@example.com"},
+    {"username": "test"}
+], "12345678");
+// user.metadata is { "email": "test@oursky.com", "username": "test" }
+// then:
+
+// - following should succeed:
+await loginWithEmail("test@oursky.com", "12345678");
+await login("test@example.com", "12345678");
+await loginWithUsername("test", "12345678");
+
+// - following should fail:
+await login({
+    "email": "test@oursky.com",
+    "username": "test"
+}, "12345678"); // cannot login with more than 2 login ID
+await signupWithEmail("test@example.com", "12345678"); // cannot reuse login ID
+await signup([
+    {"email": "test1@example.com"},
+    {"email": "test2@example.com"}
+], "12345678"); // username is not present
```

Comment:
by default username could be not present? This error only present when `username` was configured with `minimum` to 1?

Reply:
By default there can be 0 - 1 username. The section is has exactly one user name configured as mentioned at top of section, to demonstrate validation of login ID amount.